R/graph.R

```diff
@@ -31,92 +33,62 @@ graph_to_dsmj <- function(graph, dsmj_path, dsmj_name, is_directed, is_sorted=FA
   # Make named list with file names as keys and indices as values
   names(variables_indices) <- variables
 
-  # List to hold cells
-  cells_indices <- 1:nrow(edgelist_table)
-
-  # Get parameters that will be used for the cells
-  parameters <- unique(edgelist_table[["label"]])
-
-  # getCell <- function(i) {
-  #   src <- edgelist_table[["from"]][[i]]
-  #   dest <- edgelist_table[["to"]][[i]]
-  #
-  #   src_index <- variables_indices[src] - 1
-  #   dest_index <- variables_indices[dest] - 1
-  #
-  #   values <- list()
-  #   # Get all the parameter values for a specific cell
-  #   for (j in 1: length(parameters)) {
-  #     current_param <- edgelist_table[["weight"]][i]
-  #     if (current_param > 0){
-  #       values[[parameters[[j]]]] <- current_param
-  #     }
-  #   }
-  #
-  #   return(list(src=src_index, dest=dest_index, values=as.data.table(values)))
-  # }
-
-  cells <- list()
-  for (i in 1:nrow(edgelist_table)){
-    src <- edgelist_table[["from"]][[i]]
-    dest <- edgelist_table[["to"]][[i]]
+  # Group data into cells by the "from" and "to" files and get their labels and weights.
+  grouped_cells <- edgelist_table[, .(labels=list(label), weights=list(weight)),
```

Comment:
I use a group by so that I can get all of the labels and weights for each src/dest pair, and so i can use grouped_cells in my getCell and getCellReverse function